### ISSUE TITLE
On ECONNREFUSED reconnection logic doesn't kick in properly

### ISSUE THREAD

**Issue by miksago** [NodeRedis/node_redis#905]
Just hit into this and was most confused; essentially whilst there's reconnection logic, due to the `error` event being emitted on ECONNREFUSED, it fails to execute.

I had to use this code to work around it:

```
    const client = Redis.createClient(process.env.REDIS_URL);

    client.on('reconnecting', console.log.bind(console));

    // Listen to errors
    client.on('error', (err) => {
      if (err.message.indexOf('ECONNREFUSED') > -1) {
        console.log('Connection refused, should retry');
        return;
      }

      throw err;
    });
```

Otherwise on the first reconnection attempt, things will crash.

**Comment by BridgeAR** [comment]
True, this is indeed not really nice.

**Comment by gnoymmij** [comment]
I encountered the same problem. Had to fall back to 1.0.0 for now.

**Comment by BridgeAR** [comment]
So the reason why I implemented it like this originally in 2.0 was to make sure a user was able to receive the connection error just when the connection dropped.

I totally understand that this is not really nice for everyone and you might end up ignoring those errors instead.

To fix this more appropriately 2.5.0 implements the `retry_strategy`. Using this prevents the error from being emitted as the user will receive the error in the options of the retry_strategy. This function is much more powerful than all earlier reconnect settings together and being much more flexible and hopefully more intuitive than the other options.

I hope this solution is fine for you @miksago @gnoymmij
